vignettes/KEGGgraph.Rmd

```diff
@@ -0,0 +1,516 @@
+graph consists only gene products. This is especially desired when we want to
+query network characteristics of gene products. Its value is set to 'TRUE' by
+default.
+
+The following commands extract node and edge information:
+
+```{r nodeandedge, echo=TRUE, results='verbatim'}
+mapkNodes <- nodes(mapkG)
+nodes(mapkG)[1:3]
+mapkEdges <- edges(mapkG)
+edges(mapkG)[1]
+``` 
+
+Edges in KEGG pathways are directional, that is, an edge starting at node A
+pointing to node B does not guarantee a reverse relation, although reciprocal
+edges are also allowed. When listing edges, a list indexed with node names is
+returned. Each item in the list records the nodes pointed to.
+
+We can also extract the node attributes specified by KEGG with `getKEGGnodeData`:
+
+```{r keggnodedata, echo=TRUE, results='verbatim'}
+mapkGnodedata <- getKEGGnodeData(mapkG)
+mapkGnodedata[[2]]
+``` 
+
+An alternative to use `gettKEGGnodeData` is
+
+```{r keggnodedataalt, echo=TRUE, eval=TRUE}
+getKEGGnodeData(mapkG, 'hsa:5924')
+```
+
+, returning identical results.
+
+Similarly the `getKEGGedgeData` is able to extract edge information:
+
+```{r keggedgedata, echo=TRUE, results='verbatim'}
+mapkGedgedata <- getKEGGedgeData(mapkG)
+mapkGedgedata[[4]]
+``` 
+
+Alternatively the query above can be written as:
+
+```{r keggedgedataalt, echo=TRUE, eval=TRUE}
+getKEGGedgeData(mapkG,'hsa:627~hsa:4915')
+``` 
+
+For `KEGGNode` and `KEGGEdge` objects, methods are implemented to fetch their
+attributes, for example `getName`, `getType` and `getDisplayName`. Guides to use
+these methods as well as examples can be found in help pages.
+
+This case study finishes with querying the degree attributes of the nodes in the
+graph. We ask the question which nodes have the highest out- or in-degrees.
+Roughly speaking the out-degree (number of out-going edges) reflects the
+regulatory role, while the in-degree (number of in-going edges) suggests the
+subjectability of the protein to intermolecular regulations.
+
+```{r inout, echo=TRUE, results='verbatim'}
+mapkGoutdegrees <- sapply(edges(mapkG), length)
+mapkGindegrees <- sapply(inEdges(mapkG), length)
+topouts <- sort(mapkGoutdegrees, decreasing=T)
+topins <- sort(mapkGindegrees, decreasing=T)
+topouts[1:3]
+topins[1:3]
+``` 
+
+## Graph subset and merge
+
+We demonstrate the subsetting of the graph with 25 randomly chosen nodes of MAPK
+pathway graph:
+
+```{r subsetprepare, message=FALSE, results='verbatim'}
+library(Rgraphviz)
+set.seed(123)
+randomNodes <- sample(nodes(mapkG), 25)
+mapkGsub <- subGraph(randomNodes, mapkG)
+mapkGsub
+``` 
+
+The subgraph is visualized in figure [1](#fig:01), where nodes with in-degree or
+out-degree in red and others in grey [^1]. And in the example we also
+demonstrate how to convert KEGG ID into other other identifiers via the Entrez
+GeneID. More details on the conversion of IDs can be found on section [Annotation](#convertId)
```

Comment:
Let's improve the wording a little with `in the [Annotation section](#convertId).` and add a full stop at the end.